lib/authentication_with_phone_number/authentication_with_phone_number_bloc/phone_number_authentication_state.dart

```diff
@@ -0,0 +1,12 @@
+part of 'phone_number_authentication_bloc.dart';
+
+@freezed
+class PhoneNumberAuthenticationState with _$PhoneNumberAuthenticationState {
+  const factory PhoneNumberAuthenticationState({
+    @Default(FormzStatus.pure) FormzStatus status,
+    @Default(PhoneNumber.pure()) PhoneNumber phoneNumber,
+    @Default(Password.pure()) Password password,
+    @Default(NormalString.pure())NormalString verificationId,
```

Comment:
So what the normal norm and best way to ease the work you are to do is only allow the state to hold only values that you get from the use in this case, ```verificationId``` is not one of them and consequently you will be passing this value across classes and its not a good idea especially as this value is gotten only in the ```PhoneCodeSent``` callback when you call verifyPhoneNumber